### ISSUE TITLE
"A media stream ran into a fatal issue" after streaming for five to fifteen seconds

### ISSUE THREAD

**Issue by TesseractCat** [RainwayApp/bug-tracker#35]
I see this error message in the bottom right after streaming for a small amount of time, after this, all video freezes or turns black, however audio and controls still stream fine.

**Comment by TesseractCat** [comment]
Interestingly enough, it works fine when I stream through firefox v57.0 on my chromebook.

**Comment by TesseractCat** [comment]
I also see this image in the console of the client, unsure if it's related: https://i.imgur.com/ELTIrvq.png

**Comment by TesseractCat** [comment]
A solution mentioned in https://github.com/videojs/video.js/issues/1802, which is disabling the chrome flag `chrome://flags/#disable-accelerated-video-decode` fixed the issue, however probably isn't an ideal solution, since it slows down video decoding.

**Comment by justathoughtor2** [comment]
Hey all! I have the same Chromebook model as @TesseractCat and receive the same error and console log message unless I disable accelerated video decoding as indicated. This of course slows down the decoding, and thus causes other issues, but the game for the most part seems to not crash after that.

I have seen the console message in question in one other circumstance as well, and that has to do with the exact opposite scenario -- audio cuts out but video keeps playing. I haven't found even a temporary fix for that. If you need me to file that as a separate bug report, I would be more than happy to do so; the issues do seem related, though.

`message: "CHUNK_DEMUXER_ERROR_APPEND_FAILED: Got a block with a timecode before the previous block."`

**Comment by Codeusa** [comment]
This is 100% a ChromeOS/Chrome issue. we reported it [here](https://bugs.chromium.org/p/chromium/issues/detail?id=838653).

A work around would be to use Firefox. 

https://www.techrepublic.com/article/how-to-run-firefox-quantum-on-a-chromebook/

**Comment by Codeusa** [comment]
I'm not longer seeing this happen: https://twitter.com/Andrewmd5/status/1009758915305730051

**Comment by Codeusa** [comment]
Nope, I'm still seeing it too. It's still just ChromeOS. You can mitigate it by switching to Websockets from WebRTC in the settings.

**Comment by justathoughtor2** [comment]
Hey @Codeusa and @FrobtheBuilder ! I've got some good news regarding this one! Chrome OS 69 Beta has added support specifically for WebRTC hardware video decoding/encoding. (This might be in earlier versions, but if it was, I didn't see it.) Running in WebRTC mode with the flag set to its default value of Enabled seems to alleviate the errors I was seeing previously. If you guys could confirm on your end, that would be great. Maybe we can get this one closed out.

@TesseractCat Can you confirm on your ASUS C302CA model? I've only tested it on my Pixelbook. (I gave away my C302CA to a family friend recently who was in need of a computer.) You might need to be in the Beta channel to have the flag available.

**Comment by Codeusa** [comment]
We have a new update dropping soon so we can circle back after that.
